.github/workflows/jenkins-security-scan.yaml

```diff
@@ -23,15 +27,13 @@ jobs:
   scan:
     runs-on: ubuntu-latest # Provides `jq`
     steps:
-      - name: Check out plugin
-        uses: actions/checkout@v4
-
-      - name: Set up Java
-        uses: actions/setup-java@v4
+      - name: Setup Maven Action
+        uses: s4u/setup-maven-action@v1.14.0
```

Comment:
this may be a bit opinionated as this workflow also handles gradle:
https://github.com/jenkinsci/gradle-plugin/blob/ce3f778ec8f92b538563cc996d6173a60ce68908/.github/workflows/jenkins-security-scan.yml#L18

I think it'll be fine to leave it always installing maven even on gradle ones but I'm not sure if we want all of the other maven actions called.

So my recommendation would be to switch to https://github.com/stCarolas/setup-maven